unstructured_ingest/v2/processes/connectors/github.py

```diff
@@ -0,0 +1,291 @@
+from __future__ import annotations
+
+import fnmatch
+from dataclasses import dataclass, field
+from datetime import datetime
+from pathlib import Path
+from time import time
+from typing import TYPE_CHECKING, Any, Generator, List, Optional
+from urllib.parse import urlparse
+
+from pydantic import Field, root_validator
+
+from unstructured_ingest.error import SourceConnectionError, SourceConnectionNetworkError
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+    download_responses,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+CONNECTOR_TYPE = "github"
+if TYPE_CHECKING:
+    from github.Repository import Repository
+
+
+class GitHubAccessConfig(AccessConfig):
+    git_access_token: Optional[str] = Field(
+        default=None, sensitive=False, overload_name="git_access_token"
+    )
+
+
+class GitHubConnectionConfig(ConnectionConfig):
+    url: str
+    access_config: GitHubAccessConfig
+    branch: Optional[str] = Field(default=None, overload_name="git_branch")
+
+    git_file_glob: Optional[List[str]] = Field(default=None, overload_name="git_file_glob")
```

Comment:
We might not need to support this file glob parameter anymore since this is now handled by the filter process: [filter.py#L14](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/filter.py#L14)